third_party/3/pandas/core/series.pyi

```diff
@@ -149,9 +149,9 @@ class Series(base.IndexOpsMixin, generic.NDFrame):
     def apply(self, func: Function, convert_dtype: bool = ..., args: Any = ..., **kwds: Any) -> FrameOrSeries: ...
     def align(self, other: FrameOrSeries, join: JoinType = ..., axis: Optional[OneDimensionalAxisOption] = ..., level: Optional[Level] = ..., copy: bool = ..., fill_value: Optional[Scalar] = ..., method: Optional[FillMethod] = ..., limit: Optional[int] = ..., fill_axis: OneDimensionalAxisOption = ..., broadcast_axis: Optional[OneDimensionalAxisOption] = ...) -> Series: ...  # type: ignore
     @overload  #type: ignore[override]
-    def rename(self, index: Renamer = ..., *, axis: Optional[OneDimensionalAxisOption] = ..., copy: bool = ..., inplace: Literal[False] = ..., level: Optional[Level] = ..., errors: ErrorsStrategy = ...) -> Series: ...
+    def rename(self, index: Union[Renamer, Scalar] = ..., *, axis: Optional[OneDimensionalAxisOption] = ..., copy: bool = ..., inplace: Literal[False] = ..., level: Optional[Level] = ..., errors: ErrorsStrategy = ...) -> Series: ...
```

Comment:
As per documentation for Series we have:

> index : scalar, hashable sequence, dict-like or function, optional

while Renamer definition is:

> Union[Mapping[Hashable, Any], Callable[[Hashable], Hashable]]

It seems like 'hashable sequence' is not covered yet, what would you say? Also, looks like index should be Optional.

Reply:
You are right, both hashable sequences and `None` are also allowed.

`Optional` is easy to add, not sure how to do hashable sequence.
I don't see a hashable sequence defined anywhere.

I tried defining myself using `Protocol`, but this doesn't work
because `Sequence` is not a ``Protocol`

```python
from typing import Hashable, Protocol, Sequence

class HashableSequence(Hashable, Sequence[Hashable], Protocol):
    pass
```
Maybe this is one of the use cases of `Intersection` types, which are not a thing yet https://github.com/python/typing/issues/213.

The [actual implementation](https://github.com/pandas-dev/pandas/blob/6791678b19513748441c120c697ae33d74802447/pandas/core/series.py#L626) checks only for `Hashable`, So we could just use `Hashable` for hashable sequences maybe?

Or we can just do `Scalar` (and `Optional`) for now, and add hashable sequences later.
Setting a `Scalar` name is more common use,
hashable sequence names are for `MultiIndex` that is less commonly used.

Reply:
Added Optional and Hashable to the annotation to solve this.
Went for the hashable solution because any hashable value is acceptable as `Series.name` really as per the implementation. I think the documentation just says hashable sequence to mean hashable that is not Scalar.